src/webview/web_context.rs

```diff
@@ -83,33 +73,37 @@ impl WebContextImpl {
   fn set_allows_automation(&mut self, _flag: bool) {}
 }
 
-#[cfg(any(
-  target_os = "linux",
-  target_os = "dragonfly",
-  target_os = "freebsd",
-  target_os = "netbsd",
-  target_os = "openbsd"
-))]
+#[cfg(all(unix, not(target_vendor = "apple")))]
 use self::unix::WebContextImpl;
 
-#[cfg(any(
-  target_os = "linux",
-  target_os = "dragonfly",
-  target_os = "freebsd",
-  target_os = "netbsd",
-  target_os = "openbsd"
-))]
+#[cfg_attr(dox, doc(cfg(all(unix, not(target_vendor = "apple")))))]
+#[cfg(all(unix, not(target_vendor = "apple")))]
 pub mod unix {
   //! Unix platform extensions for [`WebContext`](super::WebContext).
 
+  use crate::{application::window::Window, Error};
+  use glib::FileError;
+  use std::{
+    collections::{HashSet, VecDeque},
+    rc::Rc,
+    sync::{
+      atomic::{AtomicBool, Ordering::SeqCst},
+      Mutex,
+    },
+  };
+  use url::Url;
   use webkit2gtk::{
-    ApplicationInfo, WebContext, WebContextBuilder, WebContextExt as WebkitWebContextExt,
-    WebsiteDataManagerBuilder,
+    ApplicationInfo, AutomationSessionExt, LoadEvent, SecurityManagerExt, URISchemeRequestExt,
+    UserContentManager, WebContext, WebContextBuilder, WebContextExt as WebkitWebContextExt,
+    WebView, WebViewExt, WebsiteDataManagerBuilder,
   };
 
+  #[derive(Debug)]
   pub(super) struct WebContextImpl {
-    app_info: ApplicationInfo,
     context: WebContext,
+    manager: UserContentManager,
+    webview_uri_loader: Rc<WebviewUriLoader>,
+    registered_protocols: Mutex<HashSet<String>>,
```

Comment:
Could it just use `HashSet` and make the signature of trait method accept `&mut self`?

Reply:
Yes, i was thinking that this morning. Previously it was an immutable reference because all the items were gtk specific so they had interior mutability.  `with_web_context` could probably require it mutable just fine - ill make the change now.